### ISSUE TITLE
Gather more detailed statistics for bootstrap timings

### ISSUE THREAD

**Issue by Aaron1011** [rust-lang/rustc-perf#863]
Currently, we only record the total build time for each crate in the 'bootstrap timings' section. Would it be possible to build the compiler while capturing the same level of detail as we do for other benchmaks? (e.g. self-profile information)? If this adds too much overhead, the output of `-Z time-passes` would still be very useful.

**Comment by Mark-Simulacrum** [comment]
Yeah, this should be possible - unlikely to add too much overhead. I believe the primary thing here would be to change the implementation to store these metrics. We'd also need to decide how to expose them in the UI; I'd guess the easy thing would be to just switch each bootstrap crate to act just like the normal crates, except still in a separate category... but that would need some thinking.
